odl/tomo/backends/astra_setup.py

```diff
@@ -0,0 +1,581 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helper functions to prepare ASTRA algorithms.
+
+This module contains utility functions to convert data structures from the
+`odl.tomo.geometry` representation to ASTRA's data structures, including:
+
+* volume geometries
+* projection geometries
+* data arrays
+* algorithm configuration dictionaries
+* projectors and backprojectors
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from math import sin, cos
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+try:
+    import astra
+except ImportError:
+    pass
+import numpy as np
+
+# Internal
+from odl.space.ntuples import FnVector
+from odl.discr.grid import RegularGrid
+from odl.discr.lp_discr import DiscreteLp, DiscreteLpVector
+from odl.tomo.geometry.geometry import Geometry
```

Comment:
The guideline is generally to avoid circular imports :-)

In the core part, we need to be very cautious since many things depend on many other things, and importing from anything else than the bottom-level module quickly introduces circular imports.

Here and also for the transforms as well as the solvers, we have a semi-isolation. All those parts can be seen as add-ons, which is reflected in the fact that we import the names only up to the second-highest level. Inside these "second level packages", the situation depends very much on how well-defined the order of dependency is. While this is probably totally unsettled for the `solvers`, we can safely say here that the ASTRA stuff depends on geometries, but the opposite is very unlikely. So it *should* be safe to import from `odl.tomo.geometry` directly.